### PR TITLE
storage: Assorted fixes, part 17

### PR DESCRIPTION
**Pull request by mvollmer**
The "Mount configuration" dialog unmounts the filesystem, changes fstab, and then re-mounts it.  However, if neither the mountpoint nor the options would change, the last two steps would be skipped, leaving the filesystem accidentally unmounted.

This is fixed here by not optimizing empty changes.  This is what other storage dialogs also do.

### CODE REVIEW

**Review by martinpitt — COMMENTED** [review]
Danke! This shouldn't be too hard to cover in a test?

**Comment by mvollmer** [comment]
> This shouldn't be too hard to cover in a test?

No, and it lead to a number of additional fixes... :-)

**Comment by martinpitt** [comment]
```
# test/verify/check-storage-mounting:22:1: F401 'testlib.sit' imported but unused
# test/verify/check-storage-mounting:458:1: E305 expected 2 blank lines after class or function definition, found 1
```

File: pkg/storaged/fsys-tab.jsx [hunk 1]
```diff
@@ -190,6 +192,8 @@ export function mounting_dialog(client, block, mode, forced_options) {
         let new_config = null;
         let all_new_opts;
 
+        console.log("C", JSON.stringify({ new_dir, new_opts, old_dir, old_opts }));
```
Comment (martinpitt) [inline pkg/storaged/fsys-tab.jsx:195]: debugging leftover

File: test/verify/check-storage-mounting [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 import parent  # noqa: F401
 from storagelib import StorageCase
-from testlib import test_main
+from testlib import test_main, sit
```
Comment (martinpitt) [inline test/verify/check-storage-mounting:22]: debugging leftover

File: test/verify/check-storage-mounting [hunk 3]
```diff
@@ -411,6 +411,8 @@ ExecStart=/usr/bin/sleep infinity
         m = self.machine
         b = self.browser
 
+        m.upload(["/home/mvo/work/cockpit/dist/storaged"], "/usr/share/cockpit")
```
Comment (martinpitt) [inline test/verify/check-storage-mounting:414]: debugging leftover

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Some debugging leftovers, and flake8 issues (see above), otherwise LGTM!

File: pkg/storaged/fsys-tab.jsx [hunk 4]
```diff
@@ -253,12 +257,14 @@ export function mounting_dialog(client, block, mode, forced_options) {
                                                     if (mounted_at == new_dir)
                                                         return;
                                                     return (undo()
-                                                            .then(() => client.mount_at(block, old_dir))
-                                                            .then(() => Promise.reject(error))
+                                                            .then(() => {
+                                                                if (is_filesystem_mounted)
+                                                                    return client.mount_at(block, old_dir);
+                                                            })
```
Comment (cockpituous) [inline pkg/storaged/fsys-tab.jsx:263]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18180-20230117-101932-3c5e4716-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/fsys-tab.jsx [hunk 5]
```diff
@@ -253,12 +257,14 @@
...
                                                             .catch(ignored_error => {
                                                                 console.warn("Error during undo:", ignored_error);
-                                                                return Promise.reject(error);
-                                                            }));
+                                                            })
+                                                            .then(() => Promise.reject(error)));
```
Comment (cockpituous) [inline pkg/storaged/fsys-tab.jsx:267]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18180-20230117-101932-3c5e4716-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**